File: src/main/java/com/wikia/tibia/utils/ObjectUtils.java
package com.wikia.tibia.utils;

import com.wikia.tibia.enums.YesNo;
import com.wikia.tibia.objects.Percentage;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;

import java.util.List;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class ObjectUtils {

    public static boolean isEmpty(String subject) {
        return subject == null || "".equals(subject);
    }

    public static boolean isEmpty(YesNo subject) {
        return subject == null || YesNo.EMPTY.equals(subject);
    }

    public static boolean isEmpty(Percentage subject) {
        return subject == null || Percentage.EMPTY.equals(subject);
    }

    public static boolean isEmpty(BestiaryClass subject) {
        return subject == null;
    }

    public static boolean isEmpty(BestiaryLevel subject) {
        return subject == null;
    }

    public static boolean isEmpty(Integer subject) {
        return subject == null;
    }

    public static boolean isEmpty(BookType subject) {
        return subject == null;
    }

    public static boolean isEmpty(BuildingType subject) {
        return subject == null;
    }

    public static boolean isEmpty(List<?> subject) {
        return subject == null || subject.isEmpty();
    }

    public static boolean isEmpty(ItemClass subject) {
        return subject == null;
    }

    public static boolean isEmpty(KeyType subject) {
        return subject == null;
    }

    public static boolean isEmpty(SpellType subject) {
        return subject == null;
    }

    public static boolean isEmpty(SpellSubclass subject) {
        return subject == null;
    }
}

File: src/main/kotlin/com/wikia/tibia/utils/ObjectUtils.kt
package com.wikia.tibia.utils

import com.wikia.tibia.enums.BestiaryClass
import com.wikia.tibia.enums.BestiaryLevel
import com.wikia.tibia.enums.BookType
import com.wikia.tibia.enums.BuildingType
import com.wikia.tibia.enums.ItemClass
import com.wikia.tibia.enums.KeyType
import com.wikia.tibia.enums.SpellSubclass
import com.wikia.tibia.enums.SpellType
import com.wikia.tibia.enums.YesNo
import com.wikia.tibia.objects.Percentage

object ObjectUtils {
    fun isEmpty(subject: String?): Boolean {
        return subject == null || "" == subject
    }

    fun isEmpty(subject: YesNo?): Boolean {
        return subject == null || YesNo.EMPTY == subject
    }

    fun isEmpty(subject: Percentage?): Boolean {
        return subject == null || Percentage.EMPTY == subject
    }

    fun isEmpty(subject: BestiaryClass?): Boolean {
        return subject == null
    }

    fun isEmpty(subject: BestiaryLevel?): Boolean {
        return subject == null
    }

    fun isEmpty(subject: Int?): Boolean {
        return subject == null
    }

    fun isEmpty(subject: BookType?): Boolean {
        return subject == null
    }

    fun isEmpty(subject: BuildingType?): Boolean {
        return subject == null
    }

    fun isEmpty(subject: List<*>?): Boolean {
        return subject == null || subject.isEmpty()
    }

    fun isEmpty(subject: ItemClass?): Boolean {
        return subject == null
    }

    fun isEmpty(subject: KeyType?): Boolean {
        return subject == null
    }

    fun isEmpty(subject: SpellType?): Boolean {
        return subject == null
    }

    fun isEmpty(subject: SpellSubclass?): Boolean {
        return subject == null
    }
}
